source/base/mpi.cc

```diff
@@ -812,7 +814,890 @@ namespace Utilities
 #endif
     }
 
+    template <typename T1, typename T2>
+    void
+    ConsensusAlgorithmProcess<T1, T2>::process_request(
+      const unsigned int     other_rank,
+      const std::vector<T1> &buffer_recv,
+      std::vector<T2> &      request_buffer)
+    {
+      (void)other_rank;
+      (void)buffer_recv;
+      (void)request_buffer;
+    }
+
+    template <typename T1, typename T2>
+    void
+    ConsensusAlgorithmProcess<T1, T2>::pack_recv_buffer(
+      const int        other_rank,
+      std::vector<T1> &send_buffer)
+    {
+      (void)other_rank;
+      (void)send_buffer;
+    }
+
+    template <typename T1, typename T2>
+    void
+    ConsensusAlgorithmProcess<T1, T2>::prepare_recv_buffer(
+      const int        other_rank,
+      std::vector<T2> &recv_buffer)
+    {
+      (void)other_rank;
+      (void)recv_buffer;
+    }
+
+    template <typename T1, typename T2>
+    void
+    ConsensusAlgorithmProcess<T1, T2>::unpack_recv_buffer(
+      const int              other_rank,
+      const std::vector<T2> &recv_buffer)
+    {
+      (void)other_rank;
+      (void)recv_buffer;
+    }
+
+    template <typename T1, typename T2>
+    ConsensusAlgorithm<T1, T2>::ConsensusAlgorithm(
+      ConsensusAlgorithmProcess<T1, T2> &process,
+      const MPI_Comm &                   comm)
+      : process(process)
+      , comm(comm)
+    {
+      my_rank = this_mpi_process(comm);
+      n_procs = n_mpi_processes(comm);
+    }
+
+    template <typename T1, typename T2>
+    ConsensusAlgorithm<T1, T2>::~ConsensusAlgorithm() = default;
```

Comment:
I think it is acceptable to move `default` implementations to the class declaration in the header file.